gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts

```diff
@@ -76,39 +108,75 @@ export default class ProposePayees extends SolanaCommand {
     ).every((isValid) => isValid)
 
     this.require(
-      areValidPayees || !!input.allowFundRecipient,
+      areValidPayees || !!this.input.allowFundRecipient,
       'Every payee needs to have a valid token recipient address',
     )
 
-    const proposalInfo = await program.account.proposal.fetch(proposal)
-    const payeeByTransmitter = input.operators.reduce(
-      (agg, operator) => ({
-        ...agg,
-        [new PublicKey(operator.transmitter).toString()]: new PublicKey(operator.payee),
-      }),
-      {},
-    )
-
-    // Set the payees in the same order the oracles are saved in the proposal. The length of the payees need to be same as the oracles saved
+    // Set the payees in the same order the oracles are saved in the proposal
+    // The length of the payees need to be same as the oracles saved
+    const proposal = new PublicKey(this.input.proposalId)
+    const proposalInfo = await this.program.account.proposal.fetch(proposal)
     const payees = proposalInfo.oracles.xs
       .slice(0, proposalInfo.oracles.len)
-      .map(({ transmitter }) => payeeByTransmitter[new PublicKey(transmitter).toString()])
+      .map(({ transmitter }) => this.contractInput.payeeByTransmitter[new PublicKey(transmitter).toString()])
 
-    const ix = program.instruction.proposePayees(token.publicKey, payees, {
+    const ix = this.program.instruction.proposePayees(token.publicKey, payees, {
       accounts: {
         proposal,
         authority: signer,
       },
     })
-    logger.log('Payees information:', input)
-    logger.log('Setting the following:', payees)
-
     return [ix]
   }
 
+  beforeExecute = async () => {
+    const state = new PublicKey(this.args[0])
+    let contractState = await this.program.account.state.fetch(state)
+    let numOfOracles = contractState.oracles.len.toNumber()
+
+    if (numOfOracles === 0) {
+      // handle case when no oracles were set yet
+      // iterate through oracles list in proposal
+      const proposal = new PublicKey(this.input.proposalId)
+      contractState = await this.program.account.proposal.fetch(proposal)
+    }
+
+    const payeesInContract = contractState.oracles.xs.slice(0, contractState.oracles.len.toNumber()).reduce(
+      (agg, { transmitter, payee }, idx) => ({
+        ...agg,
+        [`operator#${idx}`]: {
+          transmitter: transmitter.toString(),
+          payee: payee.toString(),
+        },
+      }),
+      {},
+    )
+
+    const proposedPayees = Object.entries(payeesInContract).reduce(
+      (agg, [key, value]) => ({
+        ...agg,
+        [key]: {
+          transmitter: (value as any).transmitter,
+          payee: this.contractInput.payeeByTransmitter[(value as any).transmitter].toString(),
+        },
+      }),
+      {},
+    )
```

Comment:
We have a small bug here, we are comparing the onchain vs onchain. I think should be:
```suggestion
    const proposedPayees = Object.entries(this.contractInput.payeeByTransmitter).reduce(
      (agg, [transmitter, payee], idx) => ({
        ...agg,
        [`operator#${idx}`]: {
          transmitter,
          payee: payee.toString(),
        },
      }),
      {},
    )
```